iceoryx_examples/request_response/README.md

```diff
@@ -0,0 +1,217 @@
+
+#include "iceoryx_hoofs/posix_wrapper/signal_watcher.hpp"
+#include "iceoryx_posh/popo/listener.hpp"
+#include "iceoryx_posh/popo/notification_callback.hpp"
+#include "iceoryx_posh/popo/server.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+```
+
+First, a callback is created that shall be called when server receives a response.
+In this case the calculation is here done too but more resource consuming tasks
+should be maybe outsourced.
+
+<!--[geoffrey][iceoryx_examples/request_response/server_cxx_listener.cpp][request callback]-->
+```cpp
+void onRequestReceived(iox::popo::Server<AddRequest, AddResponse>* server)
+{
+    while (server->take().and_then([&](const auto& request) {
+        std::cout << APP_NAME << " Got Request: " << request->augend << " + " << request->addend << std::endl;
+
+        server->loan(request)
+            .and_then([&](auto& response) {
+                response->sum = request->augend + request->addend;
+                std::cout << APP_NAME << " Send Response: " << response->sum << std::endl;
+                response.send();
+            })
+            .or_else([](auto& error) {
+                std::cout << "Could not allocate Response! Return value = " << static_cast<uint64_t>(error)
+                          << std::endl;
+            });
+    }))
+    {
+    }
+}
+```
+
+From the server port is the `take()` API used to receive the data and then the `loan()` and send `send()` calls
+to transmit the response after calculating the sum of the two numbers.
+
+
+Next, the iceoryx runtime is initialized.
+<!-- [geoffrey] [iceoryx_examples/request_response/server_cxx_listener.cpp] [initialize runtime] -->
+```cpp
+constexpr char APP_NAME[] = "iox-cpp-request-response-server-basic;
+iox::runtime::PoshRuntime::initRuntime(APP_NAME);
+```
+
+After creating the runtime, the server port is created based on a ServiceDescription.
+<!--[geoffrey][iceoryx_examples/request_response/server_cxx_listener.cpp][create server]-->
+```cpp
+iox::popo::Server<AddRequest, AddResponse> server({"Example", "Request-Response", "Add"});
+```
+
+Now we want to listen to an incoming server event and want to fire the previously created callback.
+This is done with the following call:
+<!-- [geoffrey] [iceoryx_examples/request_response/server_cxx_listener.cpp][attach listener] -->
+```cpp
+listener
+    .attachEvent(
+        server, iox::popo::ServerEvent::REQUEST_RECEIVED, iox::popo::createNotificationCallback(onRequestReceived))
+    .or_else([](auto) {
+        std::cerr << "unable to attach server" << std::endl;
+        std::exit(EXIT_FAILURE);
+    });
+```
+
+With that the preparation is done and the main thread can just sleep or do other things:
+
+<!-- [geoffrey] [iceoryx_examples/request_response/server_cxx_listener.cpp][wait for termination] -->
+```cpp
+iox::posix::waitForTerminationRequest();
+```
+
+Once the user wants to shutdown the server it finally needed to detach from the registered server event:
```

Comment:
The event is detached from the listener. 
Once the user wants to shutdown the server, the server event is detached from the listener